### PR TITLE
Various fixes and new definitions

### PR DESCRIPTION
**Pull request by Yard1**
Added definitions for most `replace_me` effects/triggers, removed the "new" files that were overwriting definitions with `replace_me`, added missing variable/array definitions, fixed various false positives and bugs.

### CODE REVIEW

File: Config/common/national_focus.cwt [hunk 1]
```diff
@@ -281,14 +298,3 @@ shared_focus = {
...
-enums = {
-	complex_enum[focuses] = {
-		path = "game/common/national_focus"
-		name = {
-			focus = {
-				id = enum_name
-			}
-		}
-	}
-}
```
Comment (Yard1) [inline Config/common/national_focus.cwt]: Not used anywhere.

File: Config/history/oobs.cwt [hunk 2]
```diff
@@ -31,24 +31,7 @@ oob = {
...
-			## cardinality = 0..1
-			amount = int
-			## cardinality = 0..1
-			requested_factories = int[1..150]
-			## cardinality = 0..1
-			progress = float[0..1]
-			## cardinality = 0..1
-			efficiency = int[0..100]
-		}
+		alias_name[effect] = alias_match_left[effect]
```
Comment (Yard1) [inline Config/history/oobs.cwt]: Instant effect supports any effect.

File: Config/common/countries.cwt [hunk 3]
```diff
@@ -32,7 +32,7 @@ enums = {
 		path = "game/common/country_tags"
 		start_from_root = yes
 		name = {
-			enum_name = scalar
+			enum_name = "scalar"
```
Comment (Yard1) [inline Config/common/countries.cwt]: Intended, otherwise it will pick up `dynamic_tags  = yes` in zz_dynamic_countries.txt

**Comment by malashin** [comment]
```
ai_will_do = {
	## cardinality = 0..inf
	factor = float
	## cardinality = 0..inf
	modifier = {
		## cardinality = 0..inf
		alias_name[trigger] = alias_match_left[trigger]
		enum[add_factor] = float
	}
}

## cardinality = 0..1
traits = {
	## cardinality = 0..inf
	<country_leader_trait> = int
}
```

I am not a fan of allowing empty scopes like this ones. Yes, it is not an error for the game engine, but why have this scope open at all then. This actually helps modders to find empty scopes then are either not filled in and left for later, or are just not needed in the first place.

We've discussed `strict` mode earlier to make different optional rules for this kind of stuff, but I don't think it is implemented. Otherwise we could think about syntax that shows warnings like it does it if you have too many of something and cardinality is lower, but the the other way around.

For example:
```
## cardinality = ~1..inf
```
If it is less then one it will show warning instead of an error.

@tboby

File: Config/events.cwt [hunk 4]
```diff
@@ -66,7 +66,7 @@ event = {
 		alias_name[trigger] = alias_match_left[trigger]
 	}
 	## cardinality = 0..1
-	fire_for_sender = no
+	fire_for_sender = bool
```
Comment (malashin) [inline Config/events.cwt]: This no/yes booleans are intetional in places that do not make sense to have it set to anything else other then the variant in the rules, the other option is default.

Comment (Yard1) [reply Config/events.cwt]: I would say that the ones I changed have reasons to be set as both yes and no, depending on what the user aims to achieve.

**Comment by Yard1** [comment]
> ```
> ai_will_do = {
> 	## cardinality = 0..inf
> 	factor = float
> 	## cardinality = 0..inf
> 	modifier = {
> 		## cardinality = 0..inf
> 		alias_name[trigger] = alias_match_left[trigger]
> 		enum[add_factor] = float
> 	}
> }
> 
> ## cardinality = 0..1
> traits = {
> 	## cardinality = 0..inf
> 	<country_leader_trait> = int
> }
> ```
> 
> I am not a fan of allowing empty scopes like this ones. Yes, it is not an error for the game engine, but why have this scope open at all then. This actually helps modders to find empty scopes that are either not filled in and left for later, or are just not needed in the first place.
> 
> We've discussed `strict` mode earlier to make different optional rules for this kind of stuff, but I don't think it is implemented. Otherwise we could think about syntax that shows warnings like it does it if you have too many of something and cardinality is lower, but the the other way around.
> 
> For example:
> 
> ```
> ## cardinality = ~1..inf
> ```
> 
> If it is less then one it will show warning instead of an error.
> 
> @tboby

As for this, I think it would be a good way to do it. Right now, they show up as errors - which they are obviously not. Had they been warnings, I would have been fine with leaving them, but errors should be left for actual errors, and not things that the engine can deal with just fine.

**Pull request merged**